chore: instructions on debuging the `gapic-generator-java` running on protobuf compiler (#1454)
googleapis/gapic-generator-java: gapic-generator-java/DEVELOPMENT.md

   ```
   and then rebuild gapic-generator-java (`mvn clean install`).

> ## Debugging the Plugin under googleapis with local gapic-generator-java (end-to-end debugging)

Author: I followed the previous section: `Running the Plugin under googleapis with local gapic-generator-java`

> ## Debugging the Plugin under googleapis with local gapic-generator-java (end-to-end debugging)

Reviewer: That section was for local gapic-generator-java. This debugging section works regardless whether it's local or not.

> ## Debugging the Plugin under googleapis with local gapic-generator-java (end-to-end debugging)

Author: done

> ## Debugging the Plugin under googleapis with local gapic-generator-java (end-to-end debugging)

Author: Right, delete it then.

To debug the local generator starting from protobuf compiler plugin

1. Add [Remote JVM debug configuration](https://www.jetbrains.com/help/idea/tutorial-remote-debug.html) and breakpoints for the local generator in Intellij.

2. In [googleapis](https://github.com/googleapis/googleapis) root directory, run the following commands:
   1. Set `JVM_DEBUG_PORT` environment variable.
       ```shell
       export JVM_DEBUG_PORT=5005

>        ```

Reviewer: Can you add why it works?

The protoc runs the "protoc-gen-java_gapic" shell script (a wrapper of java command with an option to specify JVM debugging). Because this script does not initialize `JVM_DEBUG_PORT` variable, we can pass this environment variable to the script.

Author: done

   2. Run the command to build a client library, e.g., `java-monitoring`, with `--subcommands` flag to output commands spawned by `bazel build`.
      ```shell

>       bazel build --subcommands //google/monitoring/v3:google-cloud-monitoring-v3-java

Reviewer: If you set JVM_DEBUG_PORT first, this command stops.

Author: It doesn't stop in my environment. 

In fact, environment variable set in `bazel build` has no effect on the spawned subcommand, otherwise we don't need `--subcommands`.

Reviewer: That's interesting observation. Then the order doesn't matter.

      ```

>    3. In the output, try to find the command like the below one, the `darwin_arm64-opt-exec-2B5CBBC6` part maybe different on your environment.

Reviewer: ```suggestion
   3. In the output, find the command of protoc invocation with gapic-generator-java plugin. (the `darwin_arm64-opt-exec-2B5CBBC6` part maybe different on your environment.)
```

Remove unnecesary "try to".

Explain what is that command rather than "like the below".

"the `darwin_arm64-opt-exec-2B5CBBC6` ..." is a run-on sentence.

Author: done

      ```
   4. Run the above command.
      
      It shows nothing because it's waiting for a debugger to attach.

> 3. In the local generator, start the Remote JVM Debug configuration and you can debug the generator in Intellij.

Reviewer: What does "in the local generator" mean? Can it be removed?

Author: done

> 3. In the local generator, start the Remote JVM Debug configuration and you can debug the generator in Intellij.

Reviewer: Can you add an instruction to set a breakpoint at the main function of gapic-generator-java?

Author: done

> 3. In the local generator, start the Remote JVM Debug configuration and you can debug the generator in Intellij.

Author: done

   ```
   and then rebuild gapic-generator-java (`mvn clean install`).

> ## Debugging the Plugin under googleapis with gapic-generator-java

Reviewer: ```suggestion
## Debugging the gapic-generator-java running on protoc
```

the Plugin is gapic-generator-java. It's strange.

Author: done

To debug the generator starting from protobuf compiler plugin

1. In [googleapis](https://github.com/googleapis/googleapis) root directory, run the following commands:
   1. Set `JVM_DEBUG_PORT` environment variable.
       ```shell
       export JVM_DEBUG_PORT=5005
       ```
      The proto compiler runs the `protoc-gen-java_gapic` shell script (a wrapper of java command with an option to specify JVM debugging). 

>       We set the environment variable `JVM_DEBUG_PORT` to initialize the debug mode.

Reviewer: The relationship between `protoc-gen-java_gapic` and `JVM_DEBUG_PORT` is unclear

Author: How about

We set the environment variable `JVM_DEBUG_PORT` to enable the debug mode.

Reviewer: “to enable the debug option”

(“option” is the word used in the previous sentence)

Author: done

   2. Run the command to build a client library, e.g., `java-monitoring`, with `--subcommands` flag to output commands spawned by `bazel build`.
      ```shell
      bazel build --subcommands //google/monitoring/v3:google-cloud-monitoring-v3-java
      ```

>    3. In the output, find the command of proto compiler invocation with gapic-generator-java plugin (the `darwin_arm64-opt-exec-2B5CBBC6` part maybe different on your environment).

Author: done

      ```
   4. Run the protoc command.
      
      It shows nothing because it's waiting for a debugger to attach.

> 2. In the generator, add [Remote JVM debug configuration](https://www.jetbrains.com/help/idea/tutorial-remote-debug.html) for the generator in Intellij.

Reviewer: ```suggestion
2. Add [Remote JVM debug configuration](https://www.jetbrains.com/help/idea/tutorial-remote-debug.html) for the generator in Intellij.
```

I didn't get the meaning of "In the generator".

Author: Just to guide the reader to the generator repo. Remove it also fine.

Reviewer: Then "In the IntelliJ that has gapic-generator-java"

Author: done

Author: done

3. Set a breakpoint, e.g., in the 1st line in the main method in [Main](src/main/java/com/google/api/generator/Main.java) (this line always get called).

4. Start the Remote JVM Debug configuration and you can debug the generator in Intellij.

>    ![img.png](debug.png)

Reviewer: ```suggestion
   ![IntelliJ Debug Screenshot](debug.png)
```
